src/helpers.rs

```diff
@@ -304,4 +307,40 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn eval_libc_i32(&mut self, name: &str) -> InterpResult<'tcx, i32> {
         self.eval_libc(name)?.to_i32()
     }
+
+    /// Helper function to get the `TyLayout` of a `libc` type
+    fn libc_ty_layout(&mut self, name: &str) -> InterpResult<'tcx, TyLayout<'tcx>> {
+        let this = self.eval_context_mut();
+        let tcx = &{ this.tcx.tcx };
+        let ty = this.resolve_path(&["libc", name])?.ty(*tcx);
+        this.layout_of(ty)
+    }
+
+    // Writes several `ImmTy`s contiguosly into memory. This is useful when you have to pack
+    // different values into a struct.
+    fn write_immediates(
+        &mut self,
+        ptr: &Pointer<Tag>,
+        imms: &[ImmTy<'tcx, Tag>],
+    ) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+
+        let tcx = &{ this.tcx.tcx };
+
+        let allocation = this.memory_mut().get_mut(ptr.alloc_id)?;
+        let mut offset = Size::from_bytes(0);
+
+        for imm in imms {
+            let size = imm.layout.size;
+            allocation.write_scalar(
+                tcx,
+                ptr.offset(offset, tcx)?,
+                imm.to_scalar()?.into(),
+                size,
+            )?;
```

Comment:
Quoting from the `write_scalar` docs:

> It is the caller's responsibility to check bounds and alignment beforehand.

I think it'd be better if you used the `InterpCx` methods. Generally, the `allocation` methods are low-level methods that Miri really shouldn't call, ever -- it should use the high-level place-based wrappers instead.

Reply:
hmmm but how can I take those Immediates and turn them into a tuple (or similar)?

Reply:
You don't. You still do individual writes, but use [`MPlaceTy::offset`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir/interpret/struct.MPlaceTy.html#method.offset) instead of working with bare `Pointer`.

Reply:
Oh ok! Ty